### PR TITLE
github-actions: clarify workflows permissions, set least possible privilege

### PR DESCRIPTION
**Pull request by diogoteles08**
Set top-level permissions to None on all workflows and then set per-job permissions, giving only the necessary ones. This avoids that new jobs inherit unwanted privileges.

Previously most of the workflows did not have written permissions, so their permissions were depending on the permission set as default on the Github repo settings

Discussion: https://curl.se/mail/lib-2022-11/0028.html

### CODE REVIEW

File: .github/workflows/fuzz.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ on:
     branches:
     - master
 
+permissions: {}
```
Comment (mback2k) [inline .github/workflows/fuzz.yml:16]: Please put in the same order on the job files, between concurrency and jobs sections, thanks.

**Comment by bagder** [comment]
@cmeister2 @mback2k any objections to me merging this ?

**Review by mback2k — APPROVED** [review]
LGTM

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**